opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -23,7 +23,8 @@
     these attributes can be included in the Resource.*
 
 Resource objects are created with `Resource.create`, which accepts attributes
-(key-values). Resources should NOT be created via constructor, and working with
+(key-values). Resources should NOT be created via constructor yet ResourceDetector
+instance can't use `Resource.create` to avoid infinite loops. Working with
```

Comment:
```suggestion
(key-values). Resources should NOT be created via constructor except by `ResourceDetector`
instances which can't use `Resource.create` to avoid infinite loops. Working with
```